internal/gcs/guestconnection.go

```diff
@@ -83,13 +92,14 @@ func (gcc *GuestConnectionConfig) Connect(ctx context.Context, isColdStart bool)
 
 // GuestConnection represents a connection to the GCS.
 type GuestConnection struct {
-	brdg       *bridge
-	ioListenFn IoListenFunc
-	mu         sync.Mutex
-	nextPort   uint32
-	notifyChs  map[string]chan struct{}
-	caps       schema1.GuestDefinedCapabilities
-	os         string
+	brdg           *bridge
+	ioListenFn     IoListenFunc
+	mu             sync.Mutex
+	nextPort       uint32
+	notifyChs      map[string]chan struct{}
+	caps           schema1.GuestDefinedCapabilities
+	os             string
+	initGuestState *InitialGuestState
```

Comment:
small nit: we don't really need to persist this here do we? I would prefer we just pass through the `initialGuestState` as part of the call to `connect`

Reply:
No not if we just need access to it in `connect`. I went back and forth between the two but if you prefer passing it as an argument to connect I don't have any qualms

Reply:
Changed to just passing it to `gc.connect()`